client/blocks/jetpack-benefits/site-visits.tsx

```diff
@@ -50,9 +50,12 @@ const JetpackBenefitsSiteVisits: React.FC< Props > = ( { siteId, statType, query
 		);
 	}
 
-	const countVisits = data.reduce( ( count: number, monthPeriod: { views: number } ) => {
-		return count + monthPeriod.views;
-	}, 0 );
+	const viewCounts = data as { views: number }[] | string | undefined | null;
```

Comment:
I've cast this to all sorts of things in addition to the expected data to give us guards in case the `unknown` data is not what we expect.

Reply:
we might be able to create an intermediate variable to aid in the typing. this is a trick I like to use when things are either a specific thing we want or any number of things we don't.

```ts
const data = useSelector( … );
const dataArray: { views: number }[] = Array.isArray( data ) ? data : [];

const countVisits = dataArray.reduce( ( count, { views } ) => count + views, 0 );
```

here TypeScript can infer all the types inside the `reduce` because `dataArray` is positively a `{ views: number }[]`

Reply:
Oh, that's much better. Thanks!